# Análisis de Datos de Seguros - Enfoque Step-by-Step y Caso de Negocio

Este notebook está diseñado para que resuelvas problemas paso a paso utilizando Python (usando funciones básicas como .mean(), filtros, value_counts y groupby) y que puedas comparar los resultados con el objetivo propuesto. Además, se plantea un caso de negocio ficticio en el que se asigna un gasto en marketing (CAC) para evaluar la rentabilidad y tomar decisiones estratégicas.

El objetivo es evaluar tu habilidad para:
- Explorar y analizar datos en Python.
- Resolver problemas de análisis paso a paso.
- Comparar resultados con objetivos estratégicos.
- Formular conclusiones y recomendaciones basadas en los datos.

¡Comencemos!

In [1]:
!wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip
!unzip WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip

--2025-03-15 14:37:56--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip [following]
--2025-03-15 14:37:56--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv.zip
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 352796 (345K) [

## Paso 1: Exploración Inicial del Dataset
1. Carga el dataset `WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv`.
2. Revisa las primeras filas, dimensiones y la información general del dataset.

_Observa la estructura de los datos: ¿Qué columnas tienes? ¿Se nota algún dato atípico?_

In [6]:
df.shape


(9134, 24)

In [3]:
# Cargar el dataset y ver primeras filas
import pandas as pd
import numpy as np

df = pd.read_csv('WA_Fn-UseC_-Marketing-Customer-Value-Analysis.csv')
pd.set_option('display.max_columns', None)

df.head()

Customer       State  Customer Lifetime Value Response  Coverage Education  \
0  BU79786  Washington              2763.519279       No     Basic  Bachelor   
1  QZ44356     Arizona              6979.535903       No  Extended  Bachelor   
2  AI49188      Nevada             12887.431650       No   Premium  Bachelor   
3  WW63253  California              7645.861827       No     Basic  Bachelor   
4  HB64268  Washington              2813.692575       No     Basic  Bachelor   

  Effective To Date EmploymentStatus Gender  Income Location Code  \
0           2/24/11         Employed      F   56274      Suburban   
1           1/31/11       Unemployed      F       0      Suburban   
2           2/19/11         Employed      F   48767      Suburban   
3           1/20/11       Unemployed      M       0      Suburban   
4            2/3/11         Employed      M   43836         Rural   

  Marital Status  Monthly Premium Auto  Months Since Last Claim  \
0        Married                    69                       32   
1         Single                    94                       13   
2        Married                   108                       18   
3        Married                   106                       18   
4         Single                    73                       12   

   Months Since Policy Inception  Number of Open Complaints  \
0                              5                          0   
1                             42                          0   
2                             38                          0   
3                             65                          0   
4                             44                          0   

   Number of Policies     Policy Type        Policy Renew Offer Type  \
0                   1  Corporate Auto  Corporate L3           Offer1   
1                   8   Personal Auto   Personal L3           Offer3   
2                   2   Personal Auto   Personal L3           Offer1   
3                   7  Corporate Auto  Corporate L2           Offer1   
4                   1   Personal Auto   Personal L1           Offer1   

  Sales Channel  Total Claim Amount  Vehicle Class Vehicle Size  
0         Agent          384.811147   Two-Door Car      Medsize  
1         Agent         1131.464935  Four-Door Car      Medsize  
2         Agent          566.472247   Two-Door Car      Medsize  
3   Call Center          529.881344            SUV      Medsize  
4         Agent          138.130879  Four-Door Car      Medsize

In [7]:
df.info


<bound method DataFrame.info of      Customer       State  Customer Lifetime Value Response  Coverage  \
0     BU79786  Washington              2763.519279       No     Basic   
1     QZ44356     Arizona              6979.535903       No  Extended   
2     AI49188      Nevada             12887.431650       No   Premium   
3     WW63253  California              7645.861827       No     Basic   
4     HB64268  Washington              2813.692575       No     Basic   
...       ...         ...                      ...      ...       ...   
9129  LA72316  California             23405.987980       No     Basic   
9130  PK87824  California              3096.511217      Yes  Extended   
9131  TD14365  California              8163.890428       No  Extended   
9132  UP19263  California              7524.442436       No  Extended   
9133  Y167826  California              2611.836866       No  Extended   

     Education Effective To Date EmploymentStatus Gender  Income  \
0     Bachelor           2/24/11         Employed      F   56274   
1     Bachelor           1/31/11       Unemployed      F       0   
2     Bachelor           2/19/11         Employed      F   48767   
3     Bachelor           1/20/11       Unemployed      M       0   
4     Bachelor            2/3/11         Employed      M   43836   
...        ...               ...              ...    ...     ...   
9129  Bachelor           2/10/11         Employed      M   71941   
9130   College           2/12/11         Employed      F   21604   
9131  Bachelor            2/6/11       Unemployed      M       0   
9132   College            2/3/11         Employed      M   21941   
9133   College           2/14/11       Unemployed      M       0   

     Location Code Marital Status  Monthly Premium Auto  \
0         Suburban        Married                    69   
1         Suburban         Single                    94   
2         Suburban        Married                   108   
3         Suburban        Married                   106   
4            Rural         Single                    73   
...            ...            ...                   ...   
9129         Urban        Married                    73   
9130      Suburban       Divorced                    79   
9131      Suburban         Single                    85   
9132      Suburban        Married                    96   
9133      Suburban         Single                    77   

      Months Since Last Claim  Months Since Policy Inception  \
0                          32                              5   
1                          13                             42   
2                          18                             38   
3                          18                             65   
4                          12                             44   
...                       ...                            ...   
9129                       18                             89   
9130                       14                             28   
9131                        9                             37   
9132                       34                              3   
9133                        3                             90   

      Number of Open Complaints  Number of Policies     Policy Type  \
0                             0                   1  Corporate Auto   
1                             0                   8   Personal Auto   
2                             0                   2   Personal Auto   
3                             0                   7  Corporate Auto   
4                             0                   1   Personal Auto   
...                         ...                 ...             ...   
9129                          0                   2   Personal Auto   
9130                          0                   1  Corporate Auto   
9131                          3                   2  Corporate Auto   
9132                          0                   3   Personal Auto   
9133           

## Paso 2: Análisis del Customer Lifetime Value (CLV)
1. Calcula el CLV promedio y su desviación estándar.
2. Reflexiona: ¿Qué indica una alta desviación? ¿Podrían existir segmentos de clientes con comportamientos muy distintos?

In [9]:
# Calcular el CLV promedio:
clv_mean = df['Customer Lifetime Value'].mean()
print(clv_mean)

# Calcular la desviacion estandar del CLV promedio:
clv_std = df['Customer Lifetime Value'].std()
print(clv_std)


8004.940474987081
6870.967608356924


Rta: Un CLV promedio de 8004.94 indica clientes rentables, pero una alta desviación (6870.97) revela gran variabilidad en su valor.

## Paso 3: Evaluación de la Columna Response
1. Usa `value_counts()` para ver cuántos clientes respondieron "Yes" y "No".
2. Calcula el porcentaje de respuestas "Yes".

Pregunta: ¿La proporción de "Yes" es baja? ¿Qué podría indicar esto respecto a la estrategia de respuesta?

In [10]:
df['Response'].value_counts()


Response
No     7826
Yes    1308
Name: count, dtype: int64

Rta: Una baja tasa de "Yes" (13.08%) indica oportunidades para mejorar la estrategia y optimizar la comunicación y ofertas.

## Paso 4: Análisis de la Prima Mensual (Monthly Premium Auto) por Coverage
1. Agrupa los datos por `Coverage` y calcula la media de `Monthly Premium Auto`.
2. Compara los promedios obtenidos para cada tipo de Coverage.

Pregunta: ¿Existen diferencias notables entre las coberturas? ¿Qué implicaciones tendría esto en la estrategia de precios?

In [11]:
df.groupby('Coverage')['Monthly Premium Auto'].mean()


Coverage
Basic        82.173851
Extended    103.579504
Premium     133.381068
Name: Monthly Premium Auto, dtype: float64

Rta: Las diferencias en coberturas permiten segmentar clientes; una estrategia de precios y comunicación efectiva maximizará su valor y atractivo.

## Paso 5: Análisis del Ingreso (Income) según EmploymentStatus
1. Agrupa los datos por `EmploymentStatus` y calcula el ingreso promedio.

Pregunta: ¿Qué diferencias encuentras entre los grupos? ¿Cómo podría influir esto en la segmentación de clientes?

In [12]:
df.groupby('EmploymentStatus')['Income'].mean()


EmploymentStatus
Disabled         20045.582716
Employed         56384.884521
Medical Leave    20292.770833
Retired          20554.960993
Unemployed           0.000000
Name: Income, dtype: float64

Rta: Las diferencias de ingresos según el estado laboral permiten segmentar mejor, adaptando ofertas y estrategias para aumentar ventas y satisfacción.

## Paso 6: Análisis del CLV por Estado
1. Agrupa los datos por `State` y calcula el CLV promedio.

Pregunta: ¿Qué estado muestra el mayor CLV? ¿Qué factores regionales podrían explicar esto?

In [13]:
df.groupby('State')['Customer Lifetime Value'].mean()


State
Arizona       7861.341489
California    8003.647758
Nevada        8056.706839
Oregon        8077.901191
Washington    8021.472273
Name: Customer Lifetime Value, dtype: float64

Rta: El CLV en Oregon es más alto, posiblemente por factores demográficos, económicos y culturales, brindando oportunidades para estrategias enfocadas.


## Paso 7: Análisis del Total Claim Amount según el Número de Pólizas
1. Separa a los clientes con más de 1 póliza de aquellos con 1 sola póliza.
2. Calcula el Total Claim Amount promedio para cada grupo.

Pregunta: ¿Qué diferencias encuentras y qué podrían indicar respecto al riesgo y comportamiento de reclamaciones?

In [14]:
df.groupby('Number of Policies')['Total Claim Amount'].mean()


Number of Policies
1    440.278532
2    433.610653
3    415.969821
4    428.791219
5    428.755107
6    410.962617
7    468.401546
8    445.049999
9    424.499642
Name: Total Claim Amount, dtype: float64

Rta: Las diferencias en el Total Claim Amount son mínimas, pero pueden guiar mejoras en gestión de riesgo, marketing y experiencia del cliente.

## Paso 8: Análisis del CLV según Vehicle Class
1. Agrupa los datos por `Vehicle Class` y calcula el CLV promedio para cada clase.

Pregunta: ¿Qué clase de vehículo se asocia a un mayor CLV y qué implicaciones podría tener esto en la estrategia de producto?

In [15]:
df.groupby('Vehicle Class')['Customer Lifetime Value'].mean()


Vehicle Class
Four-Door Car     6631.726607
Luxury Car       17053.348399
Luxury SUV       17122.999134
SUV              10443.511816
Sports Car       10750.989331
Two-Door Car      6671.030732
Name: Customer Lifetime Value, dtype: float64

Rta: El Luxury SUV tiene el mayor CLV y ofrece oportunidades clave para desarrollo, experiencia del cliente y marketing, impulsando rentabilidad y lealtad en la estrategia de producto.

## Paso 9: Análisis de Months Since Last Claim según Education
1. Agrupa los datos por `Education` y calcula el promedio de `Months Since Last Claim`.

Pregunta: ¿Existen diferencias notables entre los niveles educativos? ¿Qué podría indicar sobre la gestión de reclamos?

In [16]:
df.groupby('Education')['Months Since Last Claim'].mean()


Education
Bachelor                14.834789
College                 15.432301
Doctor                  15.260234
High School or Below    14.978642
Master                  15.199730
Name: Months Since Last Claim, dtype: float64

Rta: Las diferencias en "Months Since Last Claim" por nivel educativo son mínimas, sugiriendo que este factor tiene poca influencia en la gestión de reclamos.

## Paso 10: Análisis del Income según Marital Status y Gender
1. Realiza una segmentación cruzada agrupando por `Marital Status` y `Gender` para calcular el ingreso promedio.

Pregunta: ¿Qué diferencias observas? ¿Cómo podrían influir estas diferencias en la estrategia de segmentación de clientes?

In [17]:
df.groupby(['Marital Status', 'Gender'])['Income'].mean()


Marital Status  Gender
Divorced        F         37913.709450
                M         43681.934848
Married         F         44276.994962
                M         43304.110361
Single          F         23573.758120
                M         22005.776407
Name: Income, dtype: float64

Rta: Los ingresos varían según estado civil y género; los casados ganan más, lo que permite estrategias premium, mientras que los solteros requieren opciones accesibles.


## Paso 11: Distribución de la Columna Response
1. Usa `value_counts()` para mostrar cuántos clientes hay en cada grupo de Response.

Pregunta: ¿La distribución de respuestas sugiere que se debe ajustar la estrategia de comunicación?

In [18]:
df['Response'].value_counts()


Response
No     7826
Yes    1308
Name: count, dtype: int64

Rta: La baja proporción de "Yes" (13.08%) indica la necesidad de mejorar la comunicación para aumentar efectividad y engagement con los clientes.

## Paso 12: Análisis de Quejas Abiertas por Coverage
1. Agrupa los datos por `Coverage` y calcula el número promedio de quejas abiertas (`Number of Open Complaints`).

Pregunta: ¿Existen coberturas con un número de quejas significativamente mayor? ¿Qué acciones se podrían tomar?

In [19]:
df.groupby('Coverage')['Number of Open Complaints'].mean()


Coverage
Basic       0.384698
Extended    0.398979
Premium     0.333738
Name: Number of Open Complaints, dtype: float64

Rta: Las coberturas Extended y Basic tienen más quejas que Premium, aunque sin gran diferencia. Es clave investigar y mejorar procesos para reducirlas.

## Paso 13: Análisis de la Antigüedad de las Pólizas según EmploymentStatus
1. Agrupa por `EmploymentStatus` y calcula el promedio de `Months Since Policy Inception`.

Pregunta: ¿Existe alguna relación entre el estado laboral y la antigüedad de las pólizas? ¿Qué podría implicar esto en la estabilidad del cliente?

In [20]:
df.groupby('EmploymentStatus')['Months Since Policy Inception'].mean()


EmploymentStatus
Disabled         47.469136
Employed         47.800456
Medical Leave    48.141204
Retired          48.531915
Unemployed       48.747087
Name: Months Since Policy Inception, dtype: float64

Rta: Los clientes desempleados y retirados tienen mayor antigüedad en sus pólizas, lo que sugiere estabilidad y lealtad, siendo clave para estrategias de retención.

## Paso 14: Análisis del Total Claim Amount según Policy Type
1. Agrupa por `Policy Type` y calcula el Total Claim Amount promedio para cada tipo de póliza.

Pregunta: ¿Qué diferencias se observan entre los tipos de póliza y qué podrían implicar en términos de riesgo y eficiencia operativa?

In [21]:
df.groupby('Policy Type')['Total Claim Amount'].mean()


Policy Type
Corporate Auto    430.312188
Personal Auto     434.828219
Special Auto      440.472770
Name: Total Claim Amount, dtype: float64

Rta: Las pólizas tienen diferencias mínimas en Total Claim Amount, con "Special Auto" ligeramente más alto, sugiriendo mayor riesgo y oportunidad de optimización.

## Caso de Negocio: Evaluación de la Inversión en Marketing

### Contexto:
Una compañía de seguros ha asignado un presupuesto ficticio de **$5,000,000** para campañas de adquisición. El objetivo es evaluar la rentabilidad de la inversión comparando el gasto de marketing ( 50000000) (CAC) con el Customer Lifetime Value (CLV) obtenido.

### Tareas:
1. Utiliza el CLV promedio obtenido en el Paso 2.
2. Calcula el CAC promedio dividiendo el presupuesto de marketing entre el número total de clientes.
3. Compara el CAC promedio con el CLV promedio y responde:
   - ¿El CAC es inferior al CLV, lo que justificaría la inversión?
   - ¿Qué porcentaje del CLV representa el CAC?

_Reflexión estratégica: Si el CAC es significativamente menor que el CLV, la inversión en marketing es rentable. De lo contrario, se deberá replantear la estrategia de adquisición._

In [22]:
presupuesto_marketing = 5000000
clv_promedio = 8004.940
NumCli = len(df)

In [23]:
CACprmo = presupuesto_marketing / NumCli
print(CACprmo)
print(clv_promedio)

547.4052988832932
8004.94


In [24]:
CACprmo / clv_promedio * 100


6.838343558893549

Rta: El CAC es inferior al CLV, lo que justifica la inversión

### **Bonus:** Teniendo lo anterior en cual es la proporcion de clientes rentables

In [32]:
# PROPORCION DE CLIENTES RENTABLES
clientes_rentables = df[df['Customer Lifetime Value'] > cac_promedio]
numero_clientes = len(df)
proporcion_rentables = len(clientes_rentables) / numero_clientes * 100


NameError: name 'cac_promedio' is not defined

## Conclusiones y Recomendaciones

### **Bonus 2:** En no más de un parrafo mencione 3 acciones concretas para mejorar los reusltados de la empresa.


Rta: Para mejorar los resultados de la empresa, recomiendo optimizar la estrategia de segmentación, adaptando ofertas y comunicación según factores como ingresos, estado civil y nivel de cobertura para aumentar la conversión y lealtad. Además, es clave fortalecer la gestión del riesgo y la satisfacción del cliente mediante la investigación de quejas en ciertas coberturas y la optimización de procesos operativos. Finalmente, se debe mejorar la efectividad de la estrategia de comunicación para aumentar la tasa de respuestas positivas y el engagement con los clientes, asegurando una mayor retención y rentabilidad a largo plazo.